helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -762,9 +719,6 @@ public void deleteAndWaitForCompletion(String workflow, long timeout)
 
     // Deletion failed: check which step of deletion failed to complete and create an error message
     StringBuilder failed = new StringBuilder();
-    if (baseDataAccessor.exists(idealStatePath, AccessOption.PERSISTENT)) {
-      failed.append("IdealState ");
-    }
```

Comment:
Should we keep this for backward-compatibility? For legacy or existing workflows/jobs, how will their IdealStates be removed? Who will remove them?

Reply:
As @alirezazamani has also mentioned, I suppose we should keep this behavior for now. I will add a comment noting that this should be removed later.